src/builtin_math.cpp

```diff
@@ -130,6 +133,24 @@ static wcstring math_describe_error(te_error_t& error) {
     }
 }
 
+/// Return a formatted version of the value \p v respecting the given \p opts.
+static wcstring format_double(double v, const math_cmd_opts_t &opts) {
+    wcstring ret = format_string(L"%.*f", opts.scale, v);
+    // Trim trailing zeros.
+    while (!ret.empty() && ret.back() == L'0') {
```

Comment:
This might be quite a stupid question - this doesn't appear to check that the 0 is after a "."? Does it trim `1000` to `1` if scale is 0?

Reply:
Well...yes. How humiliating. Maybe less late-night coding for me.

Reply:
Answer turned out to be no, there's always a decimal  point printed, but we ought to be robust against this anyways.